app/assets/javascripts/course/components/CoursePage.js

```diff
@@ -108,9 +107,11 @@ class CoursePage extends React.Component {
     const subsection = this.state.displayedSubsection
     const component = this.state.displayedComponent
 
-    if (!component.is_complete) {
-      this.markComponentAsComplete(component)
-    } else if (!isLast(subsection.components, component)) {
+    if (subsection.is_complete) {
+      this.markSubsectionComplete(subsection)
+    }
```

Comment:
Before, there was the problem where if you finish a video or audio file and refresh or go back before pressing next, you would need to watch the video/audio all over again. I was also kind of worried cause my solution is a bit sloppy.

Now when you finish a video/audio, I call markComponentAsComplete instead of just enabling the next button which makes an API request. 

The response also tells me if the subsection is complete after completing this component. I need to somehow save that state (I think to update the sidebar? don't really remember). However, if I call markSubsectionAsComplete then, it will take me to the next component without me pressing next. 

That's why I just save the state by marking subsection.is_complete as true. Then, when next is pressed, I call markSubsectionAsComplete. It's all a bit sloppy so let me know if you can think of a better way.